### ISSUE TITLE
Funfam API requests shows job "done" but results url yields <Response [204]>.

### ISSUE THREAD

**Issue by JamesABaker** [UCL/cath-funfam-docs#2]
# Overview
Job ID: 4047bfad24537f9231124683c8601dbd

CATH version: 4.2.0

Expected behaviour: Status check reports that the job is completed. I expected to be able to access the result via an API call or a http url.

Issue: Result url only gives <Response [204]>.

# Details
In a browser, the [check status url](http://www.cathdb.info/search/by_funfhmmer/check/4047bfad24537f9231124683c8601dbd) indicates that the job is completed.
However, the view [results button url](http://www.cathdb.info/search/by_sequence?task_id=4047bfad24537f9231124683c8601dbd) does not contain the results and just displays the submission box display.

By an API request, the problem is more obvious. 
The API submitted was from python3.

Code:
```
    funfam_submission_code = "4047bfad24537f9231124683c8601dbd"
    base_url = f'http://www.cathdb.info/search/by_funfhmmer/check/{funfam_submission_code}'
    headers = {'accept': 'application/json'}
    r = requests.get(base_url, headers=headers)
    funfam_status = r.json()

    if funfam_status['success'] == 1:
        headers = {'accept': 'application/json'}
        results_url = f'http://www.cathdb.info/search/by_funfhmmer/results/{funfam_submission_code}'
        funfam_result = requests.get(results_url, headers=headers)
        funfam_api_result = funfam_result.json()

```

Error:
```
...
raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
Variables contain:
```
funfam_status=
{'success': 1, 'data': {'date_completed': '2019-04-10T17:19:29', 'status': 'done', 'worker_hostname': 'mothra.biochem.ucl.ac.uk', 'id': '4047bfad24537f9231124683c8601dbd', 'date_started': '2019-04-10T17:19:26'}, 'message': 'done'}
<Response [204]> 

funfam_result=<Response [204]> 
```

**Comment by sillitoe** [comment]
Thanks for taking the time for the excellent report.

The most likely answer is somewhat disappointing though - this is the behaviour when the search has successfully completed, but no matching results were found (the [HTTP status 204](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) signifies "No Content").
  
One of the issues with the v4.2 FunFams is coverage - when they do catch significant hits, those hits are often useful for further functional analysis (eg our CAFA results), however this accuracy comes at the cost of coverage. The good news is that we expect coverage to be improved considerably with the FunFams available in CATH v4.3 (soon my pretty, soon).

Having said all that - if you have reason to be surprised that the submitted sequence didn't return results then I can look into that as a separate issue.

**Comment by JamesABaker** [comment]
Thanks for clearing this up. So when a job completes, an empty url is made and if there are hits then this is populated with the `.json`.  I can treat empty responses as no hits. I thought 204 might be the case for uncomplete results, but it is unique to complete results with no hits:

A completed job with hits response is [<Response [200]>](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200): OK.
A completed job with no hits response is [<Response [204]>](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204): No Content.
An uncompleted job response is [<Response [500]>](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500): Internal Server Error.
An unsubmitted job/bad task id response is [<Response [404]>](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404): Not Found. No surprises there!

I look forward to 4.3!
